examples/multiple-endpoints/main.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+
+	"golang.ngrok.com/ngrok/v2"
+)
+
+func main() {
+	if err := run(context.Background()); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run(ctx context.Context) error {
+	// Create a single ngrok agent
+	agent, err := ngrok.NewAgent()
+	if err != nil {
+		return err
+	}
+
+	// Connect the agent
+	if _, err := agent.Connect(ctx); err != nil {
+		return err
+	}
+	defer agent.Disconnect()
```

Comment:
This is kind of one of the confusing parts to me about the new API where with the default agent you don't need to handle the connection/disconnection but can instead just directly call `Listen` or `Forward`. There's also no example where you use the resulting session, so having it return it and need to be ignored (in at least the ordinary case) seems a bit odd. This tripped me up a bit when configuring a custom agent.

Would it make sense to allow `Listen`/`Forward` connect if not already, and users can call `Connect` explicitly if they need to access the session information or tie the disconnection to some other signal than the default?

Reply:
yeah, we're agreed. the intention was for `WithAutoSessionManagement()` to be default true which does exactly what you would expect, but that got missed. i'll update to make it default true and update the examples to no longer show explicit connection management.

i'm going to consider removing the return session from agent.Connect() as well. i think that makes sense too. you can just access it as .Session() after Connect completes and that seems good enough